dashboard/lib/sample_data.rb

```diff
@@ -190,12 +196,58 @@ def self.create_section(options)
         # Save progress for this level if not skipping
         if best_result
           create :user_level, user: student_user, script_id: script.id,
-            level_id: script_level.level_id, attempts: 1,
+            level_id: script_level.levels.first.id, attempts: 1,
             best_result: best_result
+
+          # Create a backing channel for this level if it's a type that needs it
+          if script_level.levels.first.channel_backed?
+            ChannelToken.find_or_create_channel_token(
+              script_level.levels.first,
+              '127.0.0.1',
+              find_or_create_storage_id_for_user_id(student_user.id),
+              {
+                hidden: true
+              }
+            )
+          end
+
+          # Roll the dice to decide whether the teacher left feedback on this level
+          feedback_rand = rng.rand(100)
+          if feedback_rand < 20
+            create :teacher_feedback,
+              student: student_user,
+              teacher: options[:teacher],
+              level: script_level.levels.first,
+              comment: tiny_lipsum
+          end
         end
 
         current_level += 1
       end
     end
   end
+
+  # For the given user id, looks up the storage id, or creates a new one if the user doesn't have one.
+  def self.find_or_create_storage_id_for_user_id(user_id)
```

Comment:
any reason not to use `storage_id_for_current_user` from storage_id.rb instead of this method?